modules/api/php/endpoints/candidate/visit/visit.class.inc

```diff
@@ -171,10 +174,120 @@ class Visit extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PATCH request.
+     * For now, only support a NextStageDate argument to start next stage.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePATCH(ServerRequestInterface $request): ResponseInterface
+    {
+        $data          = json_decode((string) $request->getBody(), true);
+        $meta          = $data['Meta'] ?? [];
+        $nextStageDate = $data['NextStageDate'] ?? null;
+
+        $requiredfields = [
+            'CandID',
+            'Visit',
+            'Site',
+            'Battery',
+            'Project',
+        ];
+        $diff           = array_diff($requiredfields, array_keys($meta));
+
+        if (!empty($diff)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Field(s) missing: ' . implode(', ', $diff)
+            );
+        }
+
+        if (!$nextStageDate) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'There is no stage date specified'
+            );
+        }
+
+        $dateSegments = explode("-", $nextStageDate);
+        if (!checkdate((int)$dateSegments[1], (int)$dateSegments[2], (int)$dateSegments[0])) {
```

Comment:
Using `$date = new \DateTime($nextStageDate);` would take care of many date formats as well as validate. It throws an \Exception so you would need to put it in a try..catch.

Reply:
I rewrote this section before your comment in a slightly different way. Let me know if you agree with it.